Formula/llvm.rb

```diff
@@ -162,6 +161,115 @@ def install
     end
 
     llvmpath = buildpath/"llvm"
+    if build.stable? && build.bottle?
+      # We build LLVM a few times first for optimisations. See
+      # https://github.com/Homebrew/homebrew-core/issues/77975
+
+      # PGO build adapted from:
+      # https://llvm.org/docs/HowToBuildWithPGO.html#building-clang-with-pgo
+      # https://github.com/llvm/llvm-project/blob/33ba8bd2/llvm/utils/collect_and_build_with_pgo.py
+      # https://github.com/facebookincubator/BOLT/blob/01f471e7/docs/OptimizingClang.md
+      extra_args = [
+        "-DLLVM_TARGETS_TO_BUILD=Native",
+        "-DLLVM_ENABLE_PROJECTS=clang;compiler-rt;lld",
+      ]
+      cflags = ENV.cflags&.split || []
+      cxxflags = ENV.cxxflags&.split || []
+
+      on_macos do
+        extra_args << "-DLLVM_ENABLE_LIBCXX=ON"
+        extra_args << "-DDEFAULT_SYSROOT=#{sdk}" if sdk
+      end
+
+      extra_args << "-DCMAKE_C_FLAGS=#{cflags.join(" ")}" unless cflags.empty?
+      extra_args << "-DCMAKE_CXX_FLAGS=#{cxxflags.join(" ")}" unless cxxflags.empty?
+
+      # First, build a stage1 compiler. It might be possible to skip this step on macOS
+      # and use system Clang instead, but this stage does not take too long, and we want
+      # to avoid incompatibilities from generating profile data with a newer Clang than
+      # the one we consume the data with.
+      mkdir llvmpath/"stage1" do
+        system "cmake", "-G", "Unix Makefiles", "..",
+                        *extra_args, *std_cmake_args
+        system "cmake", "--build", ".", "--target", "clang", "llvm-profdata", "profile"
+      end
+
+      # Our just-built Clang needs a little help finding C++ headers
+      on_macos do
+        if MacOS.version <= :catalina && sdk
+          cxxflags << "-isystem#{MacOS::CLT::PKG_PATH}/usr/include/c++/v1"
```

Comment:
This is needed because, on Catalina or earlier, the `atomic` and `type_traits` headers are not in the SDK, but at `/Library/Developer/CommandLineTools/usr/include/c++/v1`. We see this on Mojave:
```
Mojave:CommandLineTools brew$ pwd
/Library/Developer/CommandLineTools
Mojave:CommandLineTools brew$ find . -name 'atomic' -print
./usr/include/c++/v1/atomic
Mojave:CommandLineTools brew$ find . -name 'type_traits' -print
./usr/include/c++/v1/experimental/type_traits
./usr/include/c++/v1/type_traits
```
On the other hand, on Big Sur:
```
❯ pwd
/Library/Developer/CommandLineTools
❯ fd '^atomic$'
usr/include/c++/v1/atomic
SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic
❯ fd '^type_traits$'
SDKs/MacOSX11.3.sdk/usr/include/c++/v1/experimental/type_traits
SDKs/MacOSX11.3.sdk/usr/include/c++/v1/type_traits
usr/include/c++/v1/type_traits
usr/include/c++/v1/experimental/type_traits
```